docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,319 @@
+
+Here is an example of the JSON definition of one such component from the Fire Aspect enchantment:
+```json
+"minecraft:post_attack": [
+    {
+        "affected": "victim",
+        "effect": {
+            "type": "minecraft:ignite",
+            "duration": {
+                "type": "minecraft:linear",
+                "base": 4.0,
+                "per_level_above_first": 4.0
+            }
+        },
+        "enchanted": "attacker",
+        "requirements": {
+            "condition": "minecraft:damage_source_properties",
+            "predicate": {
+                "is_direct": true
+            }
+        }
+    }
+]
+```
+
+Here, the Entity Effect Component is `minecraft:post_attack`. Its effect is `minecraft:ignite`, which is implemented by the `Ignite` record. This record's implementation of `EnchantmentEntityEffect#apply` sets the target entity on fire.
+
+### Vanilla Enchantment Entity Effect Component Types
+#### Defined as `DataComponentType<List<ConditionalEffect<EnchantmentEntityEffect>>>`
+- `minecraft:hit_block`: Runs an entity effect when an entity (for example, a projectile) hits a block. Used by Channeling.
+- `minecraft:tick`: Runs an entity effect each tick. Used by Soul Speed.
+- `minecraft:projectile_spawned`: Runs an entity effect after a projectile entity has been spawned from a bow or crossbow. Used by Flame.
+
+#### Defined as `DataComponentType<List<TargetedConditionalEffect<EnchantmentEntityEffect>>>`
+- `minecraft:post_attack`: Runs an entity effect after an attack damages an entity. Used by Bane of Arthropods, Channeling, Fire Aspect, Thorns, and Wind Burst.
+
+### Vanilla Enchantment Entity Effects
+- `minecraft:all_of`: Runs a list of entity effects in sequence.
+- `minecraft:apply_mob_effect`: Applies a status effect to the affected mob.
+- `minecraft:damage_entity`: Does damage to the affected entity. This stacks with attack damage if in an attacking context.
+- `minecraft:damage_item`: Damages this item's durability.
+- `minecraft:explode`: Summons an explosion. 
+- `minecraft:ignite`: Sets the entity on fire.
+- `minecraft:play_sound`: Plays a specified sound.
+- `minecraft:replace_block`: Replaces a block at a given offset.
+- `minecraft:replace_disk`: Replaces a disk of blocks.
+- `minecraft:run_function`: Runs a specified [datapack funcion].
+- `minecraft:set_block_properies`: Modifies the block state properties of the specified block.
+- `minecraft:spawn_particles`: Spawns a particle.
+- `minecraft:summon_entity`: Summons an entity.
+
+For more detail on each of these, please look at the [relevant minecraft wiki page].
+
+### Location Based Effects
+[Location Based Effect Components] are like Entity Effect Components, but they instead contain Location Based Effects. These are used for effects that need to reference specific places in the world relative to the wielder. Entity Effects are a subclass of Location Based Effects.
```

Comment:
These should come first as entity effects are a subtype. Also, restating the title of the section does not explain what location-based effects are.